src/domain/usecases/ValidateOptionSetsUseCase.ts

```diff
@@ -0,0 +1,106 @@
+import _ from "lodash";
+import { OptionSet } from "domain/entities/OptionSet";
+import { OptionValidationResult, OptionValidationRule } from "domain/entities/OptionValidationResult";
+import { OptionSetRepository } from "domain/repositories/OptionSetRepository";
+import logger from "utils/log";
+import { Async } from "domain/entities/Async";
+import { Option } from "domain/entities/Option";
+
+export class ValidateOptionSetsUseCase {
+    constructor(private readonly optionSetRepository: OptionSetRepository) {}
+
+    async execute(options: UseCaseOptions): Async<OptionValidationResult[]> {
+        logger.info("Loading optionSets...");
```

Comment:
Let's move the logger.info to the method doing the action. The cleaner we keep the public method, the better.